deploy/playbooks/03_app.yml

```diff
@@ -41,10 +49,25 @@
       when: not env_file.stat.exists
 
     - name: Start docker compose to see if everything is running
-      shell: "docker compose up -d"
+      ansible.builtin.command:
+        chdir: "{{ ansible_user_dir }}"
+        cmd: "docker compose up -d"
+      register: output
+      changed_when: output.rc != 0
+      failed_when: output.rc != 0
 
     - name: Migrate on prod
-      shell: "make prod/migrate"
+      ansible.builtin.command:
+        chdir: "{{ ansible_user_dir }}"
+        cmd: "/usr/bin/make prod/migrate"
+      register: output
+      changed_when: output.rc != 0
+      failed_when: output.rc != 0
 
     - name: Restart everything and finish
-      shell: "docker compose up -d"
+      ansible.builtin.command:
```

Comment:
Me not likey ansible.builtin.cmd. You could use `community.docker.docker_compose_v2`
E.g. https://github.com/EuroPython/infra/blob/53c2fa4d997929a095fa4b10144baa0a34b05ce7/playbooks/roles/zammad/tasks/main.yaml#L49